tests/qos/qos_sai_base.py

```diff
@@ -0,0 +1,828 @@
+            Build map of test ports index and IPs
+
+            Args:
+                request (Fixture): pytest request object
+                testPortIds (list): List of QoS SAI test port IDs
+                testPortIps (list): List of QoS SAI test port IPs
+
+            Returns:
+                testPorts (dict): Map of test ports index and IPs
+        """
+        dstPorts = request.config.getoption("--qos_dst_ports")
+        srcPorts = request.config.getoption("--qos_src_ports")
+
+        assert len(set(dstPorts).intersection(set(srcPorts))) == 0, \
+            "Duplicate destination and source ports '{0}'".format(set(dstPorts).intersection(set(srcPorts)))
+
+        assert len(dstPorts) == 3 and len(srcPorts) == 1, \
+            "Invalid number of ports provided, qos_dst_ports:{0}, qos_src_ports:{1}".format(len(dstPorts), len(srcPorts))
+
+        #TODO: Randomize port selection
+        return {
+            "dst_port_id": testPortIds[dstPorts[0]],
+            "dst_port_ip": testPortIps[testPortIds[dstPorts[0]]],
+            "dst_port_2_id": testPortIds[dstPorts[1]],
+            "dst_port_2_ip": testPortIps[testPortIds[dstPorts[1]]],
+            'dst_port_3_id': testPortIds[dstPorts[2]],
+            "dst_port_3_ip": testPortIps[testPortIds[dstPorts[2]]],
+            "src_port_id": testPortIds[srcPorts[0]],
+            "src_port_ip": testPortIps[testPortIds[srcPorts[0]]],
+        }
+
+    @pytest.fixture(scope='class', autouse=True)
+    def dutConfig(self, request, duthost):
+        """
+            Build DUT host config pertaining to QoS SAI tests
+
+            Args:
+                request (Fixture): pytest request object
+                duthost (AnsibleHost): Device Under Test (DUT)
+
+            Returns:
+                dutConfig (dict): Map of DUT config containing dut interfaces, test port IDs, test port IPs, and
+                    test ports
+        """
+        dutLagInterfaces = []
+        mgFacts = duthost.minigraph_facts(host=duthost.hostname)["ansible_facts"]
+
+        for _, lag in mgFacts["minigraph_portchannels"].items():
+            for intf in lag["members"]:
+                dutLagInterfaces.append(mgFacts["minigraph_port_indices"][intf])
+
+        testPortIds = set(mgFacts["minigraph_port_indices"][port] for port in mgFacts["minigraph_ports"].keys())
+        testPortIds -= set(dutLagInterfaces)
+        if isMellanoxDevice(duthost):
+            # The last port is used for up link from DUT switch
+            testPortIds -= set(len(mgFacts["minigraph_port_indices"]) - 1)
+        testPortIds = list(testPortIds)
+
+        # get current DUT port IPs
+        dutPortIps = {}
+        for portConfig in mgFacts["minigraph_interfaces"]:
+            portIndex = mgFacts["minigraph_port_indices"][portConfig["attachto"]]
+            if portIndex in testPortIds:
+                dutPortIps.update({portIndex: portConfig["addr"]})
+
+        testPortIps = self.__assignTestPortIps(mgFacts)
```

Comment:
Suggest to use the 'testing_port_ip_facts' module where the ip assigned is obtained from minigraph_bgp

Reply:
this is not a replacement for `testing_port_ip_facts`, it is a replacement for this module `get_ip_in_range`, the IPs in the latter module and are obtained from the the vlan IPs as could be seen [here](https://github.com/Azure/sonic-mgmt/blob/master/ansible/library/get_ip_in_range.py#L71)